File: crates/sargon-uniffi/src/hierarchical_deterministic/cap26/cap26_path/paths/identity_path.rs
use crate::prelude::*;
use sargon::IdentityPath as InternalIdentityPath;
use sargon::NewEntityPath;

#[derive(Clone, PartialEq, Eq, Hash, InternalConversion, uniffi::Record)]
pub struct IdentityPath {
    pub network_id: NetworkID,
    pub key_kind: CAP26KeyKind,
    pub index: Hardened,
}

#[uniffi::export]
pub fn new_identity_path_sample() -> IdentityPath {
    InternalIdentityPath::sample().into()
}

#[uniffi::export]
pub fn new_identity_path_sample_other() -> IdentityPath {
    InternalIdentityPath::sample_other().into()
}

#[uniffi::export]
pub fn new_identity_path(
    network_id: NetworkID,
    key_kind: CAP26KeyKind,
    index: Hardened,
) -> IdentityPath {
    InternalIdentityPath::new(network_id, key_kind, index).into()
}

File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/IdentityPath.kt
package com.radixdlt.sargon.extensions

import com.radixdlt.sargon.Cap26EntityKind
import com.radixdlt.sargon.Cap26KeyKind
import com.radixdlt.sargon.CommonException
import com.radixdlt.sargon.DerivationPath
import com.radixdlt.sargon.Hardened
import com.radixdlt.sargon.IdentityPath
import com.radixdlt.sargon.NetworkId
import com.radixdlt.sargon.cap26EntityKindToString
import com.radixdlt.sargon.newIdentityPath

fun IdentityPath.Companion.init(
    networkId: NetworkId,
    keyKind: Cap26KeyKind,
    index: Hardened
): IdentityPath = newIdentityPath(
    networkId = networkId,
    keyKind = keyKind,
    index = index
)

fun IdentityPath.asGeneral(): DerivationPath = DerivationPath.Identity(value = this)